app/models/foreman_tasks/task.rb

```diff
@@ -223,6 +227,15 @@ def to_label
       parts.join(' ').strip
     end
 
+    # Method that produces hash where key is output type (ie. stdout, stderr...) and value is output for given type
+    # So the returned hash look like {'stdout': 'normal output', 'stderr': 'error'}
+    def output_by_type
```

Comment:
Given there is only one action for which this will actually return something meaningful and that one action comes from a different plugin, shouldn't it live there?

Reply:
This method was originally in macro version in different plugin, but with the discussion with @ares I moved it to here due to better calling in templates and possibility of future usage in another places. Do you suggest to move to another place?

Reply:
well, if we want to keep it here, it needs to check whether `live_output` is defined on `main_action` right? perhaps I confused something, however I find task output in safe mode useful. However this macro is for job template output (stdout and stderr), which is only relevant for rex tasks, so it would probably make sense to live there. Not every task has stdout/stderr

Reply:
> if we want to keep it here, it needs to check whether live_output is defined on main_action right?

It needs to check whether `live_output` is defined on `main_action`, no matter where it lives, just to stay on the safe side of things.

> however I find task output in safe mode useful.

By task output you mean the thing you get by calling `Task#output` or something like REX does?